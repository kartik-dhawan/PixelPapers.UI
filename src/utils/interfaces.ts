export interface ChildrenProps {
  children: React.ReactNode
}

export type ContentfulFields = {
  [key: string]: any
}

export interface ThumbnailItem {
  fields: {
    blogMetaTitle: string
    blogTravelLocation: string
    blogType: string
    blogMetaImage: any
  }
  sys: any
}
export interface SocialMediaLink {
  id: number
  label: string
  url: string
  icon: string
}

export interface TravelBlogPost {
  blogTitle?: string
  blogMetaImage?: any
  blogMetaTitle?: string
  blogTravelLocation?: string
  blogType?: string
  blogUrlSlug?: string
  blogInitialDescription?: any
  blogThemeColor?: string
}

export interface ContentfulTravelBlog {
  fields: TravelBlogPost
  sys: any
  metadata: any
}
