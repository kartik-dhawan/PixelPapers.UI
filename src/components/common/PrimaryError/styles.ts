export const styles = {
  genericActionButtons: {
    textTransform: "capitalize",
    color: "#fefefe",
    fontWeight: "600",
    padding: 0,
    "&:hover": {
      backgroundColor: "transparent",
    },
  },
}
