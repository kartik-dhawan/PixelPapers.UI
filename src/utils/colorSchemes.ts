export const GLOBAL_COLORS = {
  TEXT_PRIMARY_DARK: "#222",
  TEXT_PRIMARY_LIGHTER: "#666",
  TEXT_PRIMARY_LIGHT: "#d9d9d9",
  BACKGROUND_PRIMARY_LIGHT: "#fefefe",
  BACKGROUND_PRIMARY_DARK: "#111",
  COMMON_FOOTER_BACKGRUND_DARK: "#383b3e",
  BLOG_SUB_COMPONENT_BACKGROUND_LIGHTER: "#efefef",
}

export const PAGE_LEVEL_COLORS = {
  PAGE_404: {
    BACKGROUND_PRIMARY_LIGHT: "#226bb152",
    TEXT_PRIMARY_LIGHT: "#444",
    TEXT_PRIMARY_DARK: "#1c354c",
  },
  TRAVEL: {
    BACKGROUND_PRIMARY_LIGHT: "#fdeded",
    BACKGROUND_PRIMARY_DARK: "#864040",
    TEXT_PRIMARY_LIGHTER: "#444",
  },
}
