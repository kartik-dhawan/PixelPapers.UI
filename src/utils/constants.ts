export const CURATED_BY_LINK_DEFAULT_TEXT = "kartik-dhawan"

export const AVERAGE_TIME_TO_READ_A_BLOG /**In minutes */ = 7
