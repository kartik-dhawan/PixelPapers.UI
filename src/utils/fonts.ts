import { DM_Sans, Anton, Libre_Barcode_39 } from "next/font/google"

export const dmSans = DM_Sans({
  weight: ["200", "300", "400", "500", "600", "700"],
  subsets: ["latin"],
  display: "swap",
  preload: true,
})

export const barcode = Libre_Barcode_39({
  weight: ["400"],
  subsets: ["latin"],
  display: "swap",
})

export const anton = Anton({
  weight: ["400"],
  subsets: ["latin"],
  display: "swap",
})
