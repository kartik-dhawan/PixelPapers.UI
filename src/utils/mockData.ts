export const TRAVEL_PAGE_THUMBNAIL_CARD_DATA = {
  blogTitle: "Top 10 Things to do on your next trip to Jibhi/Jalori Pass.",
  blogMetaImage: {
    metadata: {
      tags: [],
    },
    sys: {
      space: {
        sys: {
          type: "Link",
          linkType: "Space",
          id: "1v98waf66sfj",
        },
      },
      id: "6TVtXnjJIZ812Bxm6VBqz5",
      type: "Asset",
      createdAt: "2023-11-21T17:39:08.757Z",
      updatedAt: "2023-11-21T17:39:17.165Z",
      environment: {
        sys: {
          id: "master",
          type: "Link",
          linkType: "Environment",
        },
      },
      revision: 2,
      locale: "en-US",
    },
    fields: {
      title: "Jibhi_Jalori",
      description:
        "A picture of sunset from Raghupur fort close to jalori pass",
      file: {
        url: "//images.ctfa/IMG_3749.webp",
        details: {
          size: 71536,
          image: {
            width: 3024,
            height: 4032,
          },
        },
        fileName: "IMG_3749.webp",
        contentType: "image/webp",
      },
    },
  },
  blogMetaTitle: "Jibhi - Jalori",
  blogTravelLocation: "Himachal Pradesh",
  blogType: "Travel Itinerary",
}

export const BLOG_DATA_MOCK = {
  metadata: {
    tags: [],
  },
  sys: {
    id: "50kXqgbATvKaVDYA6WOcle",
    type: "Entry",
    createdAt: "2023-11-18T14:38:11.479Z",
    updatedAt: "2023-11-23T13:02:36.072Z",
  },
  fields: {
    blogTitle: "The Labyrith of Jodhpur",
    blogMetaImage: {
      metadata: {
        tags: [],
      },
      sys: {
        space: {
          sys: {
            type: "Link",
            linkType: "Space",
            id: "1v98waf66sfj",
          },
        },
        id: "3gVWELo7dWmYDqiDQmKd1C",
        type: "Asset",
        createdAt: "2023-11-18T14:37:49.055Z",
        updatedAt: "2023-11-21T17:40:43.224Z",
        environment: {
          sys: {
            id: "master",
            type: "Link",
            linkType: "Environment",
          },
        },
        revision: 2,
        locale: "en-US",
      },
      fields: {
        title: "Jodhpur",
        description: "Mehrangarh Fort's view from The Hosteller.",
        file: {
          url: "//images/Snapseed.webp",
          details: {
            size: 864018,
            image: {
              width: 3024,
              height: 4032,
            },
          },
          fileName: "Snapseed.webp",
          contentType: "image/webp",
        },
      },
    },
    blogMetaTitle: "The Labyrith of Jodhpur",
    blogTravelLocation: "Jodhpur, Rajasthan",
    blogType: "article",
    blogUrlSlug: "labyrinth-of-jodhpur",
    blogThemeColor: "#a19456",
    initialBlogDescription:
      "In the heart of Rajasthan, Jodhpur stands as a vibrant testament to its regal past. The majestic Mehrangarh Fort and the azure hues of the Blue City vividly recall the cultural splendor that I had the privilege to explore firsthand.",
  },
}

export const BLOG_IMAGES_ARRAY = [
  {
    sys: {
      id: "6nrhCGsnIMmlg9IKxoi0Dd",
    },
    fields: {
      title: "Serolsar Lake, Jalori pass // Image by the author",
      description: "An image taken on top of Jalori pass' Serolsar Lake.",
      file: {
        url: "//imag/IMG_5873.jpg",
        details: {
          size: 193577,
          image: {
            width: 828,
            height: 580,
          },
        },
        fileName: "IMG_5873.jpg",
        contentType: "image/jpeg",
      },
    },
  },
]

export const MOCK_ERROR = {
  name: "Err",
  message: "Error message",
}
