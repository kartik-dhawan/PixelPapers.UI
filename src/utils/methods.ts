import { GLOBAL_COLORS, PAGE_LEVEL_COLORS } from "./colorSchemes"

/**
 *
 * @param {Object} obj
 * @returns {string}
 */
export const objectToQueryString = (
  obj: any,
): string /* eslint-disable-line */ => {
  const keys = Object.keys(obj)
  const keyValuePairs = keys.map((key) => {
    return encodeURIComponent(key) + "=" + encodeURIComponent(obj[key])
  })
  return keyValuePairs.join("&")
}

/**
 *
 * @param {string} path // eg: '/travel'
 * @returns {string} // hexcode
 */
export const getThemeColorByPath = (path: string): string => {
  const colors: any = {
    "/travel": PAGE_LEVEL_COLORS.TRAVEL.BACKGROUND_PRIMARY_DARK,
  }

  return colors[path] ?? GLOBAL_COLORS.COMMON_FOOTER_BACKGRUND_DARK
}

/**
 *
 * @param {string} hex
 * @param {number} lightenValue  //number between 0-1
 * @returns
 */
export const lightenHexColor = (hex: string, lightenValue: number): string => {
  // Ensure that the input is a valid hex color code
  const hexRegex = /^#([0-9a-f]{3}|[0-9a-f]{6})$/i
  if (!hexRegex.test(hex)) {
    throw new Error("Invalid hex color code")
  }

  // Remove the "#" symbol if present
  hex = hex.replace("#", "")

  // Parse the hex color code into RGB components
  const bigint = parseInt(hex, 16)
  let r = (bigint >> 16) & 255
  let g = (bigint >> 8) & 255
  let b = bigint & 255

  // Calculate the lighter color values
  r = Math.min(255, Math.round(r + lightenValue * (255 - r)))
  g = Math.min(255, Math.round(g + lightenValue * (255 - g)))
  b = Math.min(255, Math.round(b + lightenValue * (255 - b)))

  // Convert the lighter RGB values back to hex
  const lighterHex = ((r << 16) | (g << 8) | b).toString(16).padStart(6, "0")

  // Add the "#" symbol and return the result
  return `#${lighterHex}`
}

/**
 *
 * @param {string} dateString
 * @returns {string}
 */
export const getDateFromString = (dateString?: string) => {
  if (dateString) {
    const x = Date.parse(dateString)
    const date = new Date(x)
    return `${date.toLocaleDateString("default", {
      month: "short",
    })} ${date.getDate()}`
  } else {
    const date = new Date()
    return `${date.toLocaleDateString("default", {
      month: "short",
    })} ${date.getDate()}`
  }
}

/**
 *
 * @param {number} timeInSeconds
 * @returns {number}
 */
export const secondsToMinutes = (timeInSeconds: number) => {
  return Math.ceil(timeInSeconds / 60)
}
