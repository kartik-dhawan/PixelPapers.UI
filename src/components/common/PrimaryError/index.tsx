"use client"

import { DEFAULT_EMAIL } from "@/utils/constants"
import { ErrorPage } from "@/utils/interfaces"
import { Alert, AlertTitle, Box, Button, Collapse, Stack } from "@mui/material"
import Link from "next/link"
import { useCallback, useState } from "react"
import { styles } from "./styles"

const PrimaryError = ({ error, reset }: ErrorPage) => {
  const [viewErrorDetails, setViewErrorDetails] = useState<boolean>()

  const toggleErrorDetails = useCallback(() => {
    setViewErrorDetails((prev) => !prev)
  }, [])

  return (
    <Box
      padding="1rem 1rem"
      boxSizing="border-box"
      maxWidth="100vw"
      component="section"
    >
      <Alert severity="error" variant="filled" sx={{ boxSizing: "border-box" }}>
        <AlertTitle sx={{ fontSize: "20px" }}>
          {process.env.NEXT_PUBLIC_ENVIRONMENT === "dev"
            ? error.name
            : "Internal Error"}
          <Button
            sx={{
              ...styles.genericActionButtons,
              textDecoration: "underline",
              fontWeight: "300 !important",
              marginLeft: "1rem",
            }}
            onClick={reset}
          >
            (Try again?)
          </Button>
        </AlertTitle>
        <Stack gap="1rem">
          <Box sx={{ fontSize: "18px", fontWeight: 300 }}>
            {process.env.NEXT_PUBLIC_ENVIRONMENT === "dev"
              ? error.message
              : "Something went wrong. Please try again or visit the page later."}
          </Box>
          <Collapse in={viewErrorDetails}>
            <Box
              sx={{
                marginBottom: "1rem",
                "& a": {
                  color: "#fff",
                },
              }}
            >
              {process.env.NEXT_PUBLIC_ENVIRONMENT === "dev" ? (
                error.stack
              ) : (
                <>
                  Report this issue by sending an email to{" "}
                  <Link href={`mailto:${DEFAULT_EMAIL}`}>{DEFAULT_EMAIL}</Link>
                </>
              )}
            </Box>
          </Collapse>
        </Stack>
        <Box sx={{ fontSize: "16px" }}>
          This is an error alert â€”{" "}
          <Button
            disableRipple
            sx={styles.genericActionButtons}
            onClick={toggleErrorDetails}
          >
            {viewErrorDetails ? "View Less" : "View More"}
          </Button>
        </Box>
      </Alert>
    </Box>
  )
}

export default PrimaryError
