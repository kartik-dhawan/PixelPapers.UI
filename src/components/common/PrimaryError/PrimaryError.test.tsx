import "@testing-library/jest-dom"
import renderer from "react-test-renderer"
import PrimaryError from "."
import { MOCK_ERROR } from "@/utils/mockData"

describe("Primary error component", () => {
  it("-- should match the snapshot", () => {
    const tree = renderer.create(<PrimaryError error={MOCK_ERROR} />)

    expect(tree).toMatchSnapshot()
  })
})
