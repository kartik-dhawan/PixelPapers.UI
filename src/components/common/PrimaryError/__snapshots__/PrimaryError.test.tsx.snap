// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primary error component -- should match the snapshot 1`] = `
<section
  className="MuiBox-root css-b182f3"
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-filledError MuiAlert-filled css-15i39sq-MuiPaper-root-MuiAlert-root"
    role="alert"
  >
    <div
      className="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
        data-testid="ErrorOutlineIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
    </div>
    <div
      className="MuiAlert-message css-1pxa9xg-MuiAlert-message"
    >
      <div
        className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-b0naof-MuiTypography-root-MuiAlertTitle-root"
      >
        Err
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1bsmv35-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          (Try again?)
        </button>
      </div>
      <div
        className="MuiStack-root css-8r4r5m-MuiStack-root"
      >
        <div
          className="MuiBox-root css-idktog"
        >
          Error message
        </div>
        <div
          className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
          style={
            {
              "minHeight": "0px",
            }
          }
        >
          <div
            className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
          >
            <div
              className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
            >
              <div
                className="MuiBox-root css-1rew9hp"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiBox-root css-6xix1i"
      >
        This is an error alert â€”
         
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-v1heop-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          View More
        </button>
      </div>
    </div>
  </div>
</section>
`;
